slides/r-adoption-jan2023.qmd

```diff
@@ -0,0 +1,204 @@
+
+## Goals
+
+-   **Primary Goal**: Collaborate to engineer R packages that implement important statistical methods to fill in critical gaps
+
+-   **Secondary Goal**: Develop and disseminate best practices for engineering high-quality open-source statistical software
+
+## SWE WG Activities
+
+-   First R package `mmrm` was published on CRAN in October 2022 and updated in December
+    -   We aim to estabilish this package as a new standard for fitting mixed models for repeated measures (MMRM)
+    -   We have been developing and adopting best practices for software in the `mmrm` package, and open sourced it at [github.com/openpharma/mmrm](https://github.com/openpharma/mmrm)
+    -   Currently under active development to add more features
+
+## WHY MMRM?
+
+-   Mixed Models for Repeated Measures (MMRM) is a popular choice for analyzing longitudinal continuous outcomes in randomized clinical trials
+-   No great R Package - initially though that the MMRM problem was solved by using a combination of `lme4` and `lmerTest`
+-   Learned that this approach failed on Ophthalamology data (slow, did not converge
+-   `nlme` does not give Satterthwaite adjusted degrees of freedom, has convergence issues, and with `emmeans` it is only approximate
+-   tried to extend `glmmTMB` to calculate Satterthwaite adjusted degrees of freedom
+
+## Idea
+
+-   Because `glmmTMB` is always using a random effects representation, we cannot have a real unstructured model (uses $\sigma = \varepsilon > 0$ trick)
+-   We only want to fit a fixed effects model with a structured covariance matrix for each subject
+-   The idea is then to use the Template Model Builder (`TMB`) directly - as it is also underlying `glmmTMB` - but code the exact model we want
+-   We do this by implementing the log-likelihood in `C++` using the `TMB` provided libraries
+
+## Advantages of `TMB`
+
+-   Fast `C++` framework for defining objective functions (`Rcpp` would have been alternative interface)
+-   Automatic differentiation of the log-likelihood as a function of the variance parameters
+-   We get the gradient and Hessian exactly and without additional coding
+-   This can be used from the R side with the `TMB` interface and plugged into optimizers
+
+## What's next with {mmrm} and SWE WG
+
+-   Prepare public training materials to disseminate best practice for software engineering in the Biostatistics community
+    -   At the beginning of February, a face-to-face workshop will take place in Basel, Switzerland with a focus on open-source software for clinical trials
+    -   Organize conference sessions with a focus on statistical software engineering at CEN, JSM and ASA/FDA Workshop
+    -   Video clips in preparation with a focus on best practices for software engineering
+
+## Why it's not just another package
+
+-   Ongoing maintenance and support from the pharmaceutical industry that is supported by American Statistical Association
```

Comment:
not sure if ASA really supports the package as such ... but I would highlight that multiple companies are supporting it via developers